config/etc/nginx/nginx.conf

```diff
@@ -46,7 +46,19 @@ http {
       ## See the map directive above where this variable is defined.
       add_header 'Docker-Distribution-Api-Version' $docker_distribution_api_version always;
 
-      proxy_pass                          http://registry:5000; #${PORT};
+      proxy_pass                          http://registry:5000;
+      proxy_set_header  Host              $http_host;   # required for docker client's sake
+      proxy_set_header  X-Real-IP         $remote_addr; # pass on real client's IP
+      proxy_set_header  X-Forwarded-For   $proxy_add_x_forwarded_for;
+      proxy_set_header  X-Forwarded-Proto $scheme;
+      proxy_read_timeout                  900;
+    }
+
+    # TODO: Not sure if that works, need for research about it
```

Comment:
What does this comment refer to?  This setup seems to work.  Should we remove it?

Reply:
Maybe here.

Reply:
Yeah, removed